### PR TITLE
Adding pixeloven and cli

### PR DESCRIPTION
**Pull request by ductiletoaster**
This PR is the first of several to move all CLI functionality into another project. I'll like follow up with some changes to reduce the amount of configuration needed. The goal would make it so the CLI project has sensible defaults reducing/eliminating the need for configuration on our side.

I also revamped the CLI and commit hooks to do a lot of the formatting for you. Tslint, stylelint and prettier together should handle everything. 

TypeScript should also show a code frame for linter errors. (eventually want to do this for compiling too)

New CLI project can be found here.
https://github.com/pixeloven/pixeloven

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +135,9 @@ With all of the items below we need to consider whether we want to maintain a si
     * Use reselect to prevent immutableJS for breaking into components
 * Look into MVCish frameworks for the client server
 * Does it still make sense to use `yarn`? `npm`? Or another alternative?
+* Get prettier adn stylelint working together https://www.npmjs.com/package/prettier-stylelint
```
Comment (jimmynotjim) [inline README.md]: Nit `adn` -> `and`

File: package.json [hunk 2]
```diff
@@ -83,161 +74,57 @@
...
+    "start": "run-p start:app start:story",
+    "start:app": "pixeloven-tasks serve",
+    "start:story": "start-storybook --quiet -s ./public -p 9001 -c .storybook",
+    "test": "pixeloven-cli test",
+    "test:watch": "pixeloven-cli test:watch"
   },
   "lint-staged": {
-    "lib/**/*.{ts,tsx}": [
-      "yarn lint-app-fix",
+    "src/**/*.{scss,ts,tsx}": [
```
Comment (jimmynotjim) [inline package.json]: Why do we need lint these files as a group here and split out by file type below?

Comment (ductiletoaster) [reply package.json]: I had them all together but then realized that husky pipes the file names to the script and so by splitting them we can better optimize the jobs.

Comment (jimmynotjim) [reply package.json]: Do we need them grouped at all then or can this first item be removed?

**Pull request merged**